lib/SILOptimizer/SILCombiner/SILCombine.cpp

```diff
@@ -152,6 +152,8 @@ bool SILCombiner::doOneIteration(SILFunction &F, unsigned Iteration) {
 
   SILCombineCanonicalize scCanonicalize(Worklist);
 
+  bool shouldTryOptimizeStringSwitch = false;
```

Comment:
Is this a good way to do this?

Reply:
IMO, it would be much better to create a separate function pass. It does not make sense to pack everything into SILCombine. Especially this optimization, because it deals with the whole function anyway.

Reply:
What do you mean a separate function pass? I'm not super familiar with the SILOptimizer's structure.

Reply:
@eeckstein do you mean that I should create a `SILModuleTransform` class and add it to `Passes.def`?

Reply:
No, it would be a SILFunctionTransform. To create it, you'll need to create a source file for it, register the pass in Passes.def and insert it somewhere in the pass pipeline in PassPipeline.cpp. As an example you can take a look at https://github.com/apple/swift/pull/26803/files

Reply:
Great, thanks. Updated into a `SILFunctionTransform`.

Reply:
Hmm... After this change, some static string arrays turn into outlined variables and some that used to be outlined variables stop being outlined variables. Any idea why?